Makefile

```diff
@@ -169,12 +172,21 @@ $(TERRAFORM):
 	@rm -fr $(TOOLS_HOST_DIR)/tmp-terraform
 	@$(OK) installing terraform $(HOSTOS)-$(HOSTARCH)
 
-$(TERRAFORM_PROVIDER_SCHEMA): $(TERRAFORM)
+$(TERRAFORM_PROVIDER):
+	@$(INFO) installing terraform provider $(SAFEHOST_PLATFORM)
+	@mkdir -p $$(dirname $(TERRAFORM_PROVIDER))
+	@curl -fsSL https://github.com/upbound/terraform-provider-aws/releases/download/$(TERRAFORM_PROVIDER_RELEASE)/terraform-provider-aws_$(TERRAFORM_PROVIDER_RELEASE)_$(SAFEHOST_PLATFORM).zip -o $(TERRAFORM_PROVIDER).zip
+	@unzip $(TERRAFORM_PROVIDER).zip -d $$(dirname $(TERRAFORM_PROVIDER))
+	@rm -fr $(TERRAFORM_PROVIDER).zip
+	@$(OK) installing terraform provider $(SAFEHOST_PLATFORM)
+
+$(TERRAFORM_PROVIDER_SCHEMA): $(TERRAFORM) $(TERRAFORM_PROVIDER)
 	@$(INFO) generating provider schema for $(TERRAFORM_PROVIDER_SOURCE) $(TERRAFORM_PROVIDER_VERSION)
 	@mkdir -p $(TERRAFORM_WORKDIR)
 	@echo '{"terraform":[{"required_providers":[{"provider":{"source":"'"$(TERRAFORM_PROVIDER_SOURCE)"'","version":"'"$(TERRAFORM_PROVIDER_VERSION)"'"}}],"required_version":"'"$(TERRAFORM_VERSION)"'"}]}' > $(TERRAFORM_WORKDIR)/main.tf.json
```

Comment:
nit: Currently when you examine the provider block, it looks like we are using a vanilla Terraform AWS provider (because the URI is till `hashicorp/aws:v5.31.0`) however, we are using a custom built binary. We may, in a future PR, consider making this more explicit by changing the Terraform provider URI.